packages/auth-providers/netlify/api/src/__tests__/netlify.test.ts

```diff
@@ -1,22 +1,27 @@
 import type { APIGatewayProxyEvent, Context as LambdaContext } from 'aws-lambda'
 import jwt from 'jsonwebtoken'
+import { vi, beforeAll, afterAll, test, expect } from 'vitest'
 
 import { authDecoder } from '../decoder'
 
-jest.mock('jsonwebtoken', () => {
-  const jsonwebtoken = jest.requireActual('jsonwebtoken')
+vi.mock('jsonwebtoken', async (importOriginal) => {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+  const originalJWT = await importOriginal<typeof import('jsonwebtoken')>()
 
   return {
-    ...jsonwebtoken,
-    verify: jest.fn(),
-    decode: jest.fn((token: string) => {
-      const exp =
-        token === 'expired-token'
-          ? Math.floor(Date.now() / 1000) - 3600
-          : Math.floor(Date.now() / 1000) + 3600
-
-      return { exp, sub: 'abc123' }
-    }),
+    ...originalJWT,
+    default: {
+      verify: vi.fn(),
+      decode: vi.fn((token: string) => {
+        const exp =
+          token === 'expired-token'
+            ? Math.floor(Date.now() / 1000) - 3600
+            : Math.floor(Date.now() / 1000) + 3600
+
+        return { exp, sub: 'abc123' }
+      }),
+    },
+    TokenExpiredError: originalJWT.TokenExpiredError,
```

Comment:
This one I had to explicitly state and doesn't appear to be covered by the general spread a couple lines above.

Reply:
Looks like this is available on `originalJWT.default`, but `originalJWT.default` is untyped. So we could spread that but then we'd have to have a `// @ts-expect-error`